proposals/billboards-xformOp/README.md

```diff
@@ -0,0 +1,320 @@
+
+`affixPosition`, `affixScale` and `align` are **compound** since they all require a set of
+attributes (instead of single one) to compute the transformation matrix. All three
+XformOp are **extrinsic** as they all request the current viewing states (such as
+the view matrix and the projection matrix) to compute the local transformation of the
+billboarded primitives (again, a transformable in USD).
+
+XformOps `affixPosition` and `affixScale` can specify the target position and size
+in a selected coordinate space:
+- `ndc`: coordinate is in the normalized device space ([-1.0, 1.0]x[-1.0, 1.0])
+- `view`: coordinate is in the view space, where camera is at (0,0,0) and faces the x-y plane.
+- `screen`: coordinate is in the screen space ([0, width-1]x[0, height-1] where width and
+height are in pixels).
+
+XformOp `affixPosition` consists of two attribute fields to affix the transformable to the
+specified screen location:
+- `affixPosition:anchor`: source location that is specified in the model space.
+- `affixPosition:anchorage`: target location that is specified in the selected
+coordinate space (`ndc`, `view`, or `screen`).
+
+`affixPosition` constructs the local transformation that affixes `anchor` to `anchorage`
+on the screen.
+
+XformOp `affixScale` consists of three attribute fields to affix the screen size of the
+transformable.
+- `affixScale:baseSize`: the unit size in the model space.
+- `affixScale:scaleSize`: the equivalent unit size in the selected coordinate space (
+`ndc`, `view`, or `screen`).
+- `affixScale:pivot`: the center position of the scaling in the model space.
+
+`affixScale` constructs the local transformation that scales the transformable to
+match the model-space unit size of `baseSize` to `scaleSize` in the designated space.
+
+XformOp `align` fixes the alignment with a reference plane that is orthogonal to the
+front normal of the transformable. Three types of the reference plane are supported:
+- `screen`: front normal is orthogonal to the screen plane (i.e. the near plane of the
+viewing frustum).
+- `viewer`: front normal points to the camera location.
+- `plane`: front normal is orthogonal to a custom plane.
+
+`align` constructs the local transformation that maintains the orientation of the
+transformable's front facing. It consists of one optional attribute field for the
+`plane` reference plane:
+- `align:plane`: a 3D plane defined by the authored normal of the plane.
+
+XformOps `affixPosition`, `affixScale`, and `align` can be turned off with authored
+value of `off`. This `off` mode allows artist and developers to disable the billboards
+transformation with overrides.
+
+All billboards XformOps are acceptable Ops in `xformOpOrder`. Billboards XformOps can
+combine with the existing XformOps in USD. This allows the use cases such as rendering a
+fixed-size axis-tripod at a fixed screen location. The axis-tripod can, of course, synchronize
+its rotation with the trackball action of the camera.
```

Comment:
Your point on the pricy decompositon of current transformation matrix (CTM)  is valid. When I worked on the prototype, I have difficulties to extract the rotation reliablely. Without certain pre-coditions of the transformation matrix, I did not find a reliable way to decompose it .

This part of proposal is confusing and probably misleading. I'll make changes to it.

Here is what I think at this moment:
- Billboards xformOp constructs a local matrix without considering CTM. The billboards matrix will be applied to CTM just like what the current xformOps are doing.
- In addition to `resetXformStack`, we might want to add new `resetOp` such as `resetTranslation', `resetScale` and `resetRotation`. Such `resetOp` may not work if CTM is degenrated unfornately ;(
- Users should optionaly prepend a `resetXformStack` or a `resetOp` before billboards xformOps.

I would prefer to make the `reset` explict instead of having it implictly reset inside. Baking the `reset` to billboards does not seem to be a good idea. It is not flexible and can impact the perpormance even for use cases that the reseting is not required.